src/components/CreateVmWizard/steps/Storage.js

```diff
@@ -226,11 +226,24 @@ class Storage extends React.Component {
           const {
             storageDomainId: id,
             storageDomain: sd,
+            sdErrFlag,
           } = rowData
 
-          // TODO: if the disk's storage domain is not available to the user, highlight it
-          // TODO: like a validation error and force editing the SD before allowing the user to move
-          // TODO: forward to the next wizard step
+          if ((rowData.isFromTemplate && !sd.isOk) || sdErrFlag) {
+            const { storageDomains, dataCenterId } = props
+            const storageDomainList = createStorageDomainList(storageDomains, dataCenterId, true)
+            if (!sd.isOk) {
+              storageDomainList.unshift({ id: '', value: `-- ${msg.createVmStorageSelectStorageDomain()} --` })
+            }
+            return <SelectBox
+              id={`${idPrefix}-${rowData.id}-storage-domain-edit`}
+              items={storageDomainList}
+              selected={sd.isOk ? rowData.storageDomainId : ''}
+              onChange={value => {
+                this.props.onUpdate({ 'update': { ...rowData, storageDomainId: value, sdErrFlag: true } })
```

Comment:
Hi Radek,
let's take a look on the storage domain `valueView` function:
https://github.com/oVirt/ovirt-web-ui/blob/3a5e6bf5ca0a8c76ac961fcb67435efe68e13512/src/components/CreateVmWizard/steps/Storage.js#L225-L255
I want to focus on the use of the `sdErrFlag` - line 232, the if statement:
if we have case of invalid storage domain the `sd.isOk` will be `undefined` otherwise it will be true even it was invalid first and then the user changed it, if we don't use the `sdErrFlag` in the if statement the user could change it once and then it will be unchangeable. so to avoid this I used a dedicated flag to signify that the template defined disk was invalid for the current user, so this flag should be always true if the disk's SD was invalid at first, otherwise it will change once.
that's the purpose of using always `true` for this flag :)

Reply:
Thanks for explanation! I would suggest renaming it to sth like `createdFromInvalidTemplate` and adding some comments in the code to explain this use case (that user should always be able to change it).
How about remaining points?

> It seems we should trigger the validations and set here also the valid flag (otherwise it will be always true).
> It seems we don't need to pass update prop name as a string.

Reply:
> ...the user could change it once and then it will be unchangeable...

I wonder why it was implemented like that, originally. I think we should solve this the other way than adding an extra flag which will always be set to the same value. But I feel I still don't understand all the details so maybe I just don't see the reason why you've chosen to solve the problem of changing SD by adding such a flag :)

Reply:
Hi Radek,
the validations are in the CreateVmWizard component and the update prop is not string anymore.
Thanks.